File: ios/Reguerta/Reguerta/Data/Shifts/ChainedShiftRepository.swift
import Foundation

struct ChainedShiftRepository<Primary: ShiftRepository, Fallback: ShiftRepository>: ShiftRepository {
    let primary: Primary
    let fallback: Fallback

    func allShifts() async throws -> [ShiftAssignment] {
        try await primary.allShifts()
    }

    func upsert(shift: ShiftAssignment) async throws -> ShiftAssignment {
        try await primary.upsert(shift: shift)
    }
}

File: android/Reguerta/app/src/main/java/com/reguerta/user/data/shifts/ChainedShiftRepository.kt
package com.reguerta.user.data.shifts

import com.reguerta.user.domain.shifts.ShiftAssignment
import com.reguerta.user.domain.shifts.ShiftRepository

class ChainedShiftRepository(
    private val primary: ShiftRepository,
    @Suppress("UNUSED_PARAMETER") fallback: ShiftRepository,
) : ShiftRepository {
    override suspend fun getAllShifts(): List<ShiftAssignment> = primary.getAllShifts()
}
